lib/package-managers/npm.js

```diff
@@ -50,7 +67,26 @@ function view(packageName, field, currentVersion) {
     if (currentVersion && (!semver.validRange(currentVersion) || versionUtil.isWildCard(currentVersion))) {
         return Promise.resolve();
     }
-    const options = {agent, allVersions: true};
+    const options = {allVersions: true};
+
+    if (proxyAgent && noProxy !== true) {
```

Comment:
Prefer `proxyAgent && !noProxy`

Reply:
That wouldn't have worked in this case, because the intention was to exactly check for `true`, because `noProxy` could have been an array too. A bit confusing; I've tried to address that.